ganttproject/src/main/java/net/sourceforge/ganttproject/chart/ChartModelBase.java

```diff
@@ -190,7 +194,55 @@ public boolean apply(Task input) {
   private final GPOptionGroup myTimelineLabelOptions;
 
   private final FontOption myChartFontOption;
-  private final BooleanOption myWeekNumberOption;
+
+  private final DefaultEnumerationOption myWeekNumberOption;
+  private TimeFormatters.ObservableProperty<java.util.function.Function<Date, Integer>> myWeekNumProperty;
+
+  private final static Function<Date, Integer> US_WEEK_NUMBERING = date -> {
```

Comment:
Let's translate these functions to Kotlin and place in `WeekNumbering.kt` nearby?